lib/Config/Migrator.php

```diff
@@ -448,6 +449,26 @@ function statisticsForms() {
     return $this->sqlify(__FUNCTION__, $attributes);
   }
 
+  function statisticsWoocommerceOrders() {
+    $attributes = array(
+      'id int(11) unsigned NOT NULL AUTO_INCREMENT,',
+      'newsletter_id int(11) unsigned NOT NULL,',
+      'subscriber_id int(11) unsigned NOT NULL,',
+      'queue_id int(11) unsigned NOT NULL,',
```

Comment:
Do we need to store all the information here? If we store `click_id` we can get `queue_id`, `subscriber_id` and `newsletter_id` from there, can't we?

Reply:
At the moment we need `newsletter_id` and `queue_id` to compute & show stats. I was proposing a simplification that would use just `newsletter_id` but that's not part of this task. So `subscriber_id` may not be necessary here but it's consistent with other stats and might be useful for debugging... what do you think?

Reply:
> At the moment we need newsletter_id and queue_id to compute & show stats.

I don't understand why do we need it if we can join clicks and get `newsletter_id` and `queue_id` that way.

Reply:
That could make showing stats on mail listing page much slower. For each row you'd need a JOIN to clicks table. Yes, JOINs are usually faster than 1 request per row, especially in Postgres with hash joins, etc., but the overhead is still significant. Also, if stats should ever use indexes only, there's no way to create compound indexes over two tables.

Reply:
That said, I am not sure if I should remove `subscriber_id`. We do use it for other stats like https://github.com/mailpoet/mailpoet/blob/master/lib/Models/StatisticsClicks.php#L35 and it seems to be for some exporter - in that case I suspect we'll need it for this new stats as well.